packages/reactive-element/src/decorators/query-assigned-elements.ts

```diff
@@ -67,11 +67,18 @@ export function queryAssignedElements(options?: QueryAssignedElementsOptions) {
         const slotSelector = `slot${slot ? `[name=${slot}]` : ':not([name])'}`;
         const slotEl =
           this.renderRoot?.querySelector<HTMLSlotElement>(slotSelector);
-        const elements = slotEl?.assignedElements(options) ?? [];
+        // HTMLSlotElement.assignedElements would be preferred here, however
+        // there isn't a good path forward regarding browser polyfill support.
+        const elements = slotEl?.assignedNodes(options) ?? [];
```

Comment:
What do you think about effectively including a tiny module-scoped polyfill directly in this file, so that we'll still use built-in `assignedElements` when available?

Maybe something like this?


```ts
const slotAssignedElements =
  HTMLSlotElement.prototype.assignedElements ??
  function slotAssignedElementsPolyfill(
    this: HTMLSlotElement,
    opts: AssignedNodesOptions
  ) {
    return this.assignedNodes(opts).filter(
      (node) => node.nodeType === Node.ELEMENT_NODE
    );
  };

...

const elements =
  slotEl !== null ? slotAssignedElements.call(slotEl, options) : [];
```

It has the advantage of using the native method when available instead of the filter. Presumably this is faster... though I don't know for sure.

It does mean that when a selector is provided, and the polyfill is used, there will be 2 filters instead of 1. But probably preferring slightly better performance on newer browsers is the direction we should lean in?

Reply:
Would the `HTMLSlotElement` exist on IE11, for example when shadyDOM is in use?

Reply:
Good question, I think not reading https://github.com/webcomponents/polyfills/blob/0d119eadcf382f4e5ed865c5b96e87c9350e2d1e/packages/shadydom/src/patch-prototypes.js#L72, but I'm not sure...

@bicknellr @kevinpschaaf can you confirm?

If not I suppose you can do `window?.HTMLSlotElement?.prototype?.assignedElements` instead?

Reply:
Done! Great suggestion to measure bundle size!

Reply:
I'm not sure, I assumed Shady DOM would need to create `HTMLSlotElement` if it doesn't exist but I can't find where it does that. cc @sorvell

Reply:
I can also add that depending on HTMLSlotElement directly fails some tests. Using `window.HTMLSlotElement?.prototype.assignedElements` works.